src/cf.data.pre

```diff
@@ -2851,7 +2851,39 @@ DOC_START
 	based on the username or source address of the user making
 	the request.
 
-	tcp_outgoing_address ipaddr [[!]aclname] ...
+	tcp_outgoing_address <address> [[!]aclname] ...
+
+	The required address parameter must be one of the following:
+
+		IP: A literal IP address.
+
+		domain: A domain name that Squid attempts to resolve once, at each
+			(re)configuration time, using getaddrinfo(3) or a similar blocking
+			service. DNS resolution errors are fatal.
+
+		match_client_tcp_dst: This keyword instructs Squid to postpone
+			outgoing address determination until this directive is actually
+			used for forwarding a client request. At that time, Squid attempts
+			to set the outgoing IP address to the TCP client destination
+			address. Squid silently skips these rules if it cannot determine
+			the corresponding client request or its destination address at
+			tcp_outgoing_address application time.
+
+			For example, if Squid listens on both 10.0.0.1 and 10.0.0.2, but
+			accepts an HTTP request at 10.0.0.2, then matching a
+			tcp_outgoing_address rule with this keyword will set the source
+			address of the outgoing request connection to 10.0.0.2.
+
+		other: Squid also accepts a few currently undocumented keywords.
```

Comment:
I did not want to document the other accepted keywords (i.e. "any_addr", "no_addr", and "full_mask"), but I would not object to others doing that. I am also OK with removing this line if others do not like it.

@DanWritesCode, I suspect this may be a controversial topic, and it is out of this PR scope, so I do not recommend documenting those keywords unless @yadij asks you to do that. Removing this (and any other) line if you do not like it is fine, of course.

Reply:
One option to consider mentioning here is that "other" can include "any_addr", "no_addr", and "full_mask". This may prompt someone to look into how those options work with this setting, if there is any interest in it, without us having to spend time documenting it.